src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -584,33 +585,43 @@ public void removeMinimumStock(final ItemStack itemStack)
     public void onColonyTick(final IColony colony)
     {
         super.onColonyTick(colony);
-        final Collection<IToken<?>> list = getOpenRequestsByRequestableType().getOrDefault(TypeToken.of(Stack.class), new ArrayList<>());
 
-        for (final Map.Entry<ItemStorage, Integer> entry : minimumStock.entrySet())
+        if (colony.getWorld().getChunkProvider().isChunkLoaded(new ChunkPos(getPosition().getX() >> 4, getPosition().getZ() >> 4)))
         {
-            final ItemStack itemStack = entry.getKey().getItemStack().copy();
-            final int count = InventoryUtils.getItemCountInProvider(getTileEntity(), stack -> !stack.isEmpty() && stack.isItemEqual(itemStack));
-            final int delta = entry.getValue() * itemStack.getMaxStackSize() - count;
-            final IToken<?> request = getMatchingRequest(itemStack, list);
-            if (delta > 0)
+            final Collection<IToken<?>> list = getOpenRequestsByRequestableType().getOrDefault(TypeToken.of(Stack.class), new ArrayList<>());
+
+            for (final Map.Entry<ItemStorage, Integer> entry : minimumStock.entrySet())
             {
-                if (request == null)
+                final ItemStack itemStack = entry.getKey().getItemStack().copy();
+
+                if (itemStack.isEmpty())
                 {
-                    itemStack.setCount(Math.min(itemStack.getMaxStackSize(), delta));
-                    final Stack stack = new Stack(itemStack);
-                    if (getMainCitizen() != null)
-                    {
-                        getMainCitizen().createRequestAsync(stack);
-                    }
-                    else
+                    continue;
+                }
+
+                final int count = InventoryUtils.hasBuildingEnoughElseCount(this, new ItemStorage(itemStack), entry.getValue() * itemStack.getMaxStackSize());
+                final int delta = (entry.getValue() * itemStack.getMaxStackSize()) - count;
+                final IToken<?> request = getMatchingRequest(itemStack, list);
+                if (delta > 0)
+                {
+                    if (request == null)
                     {
-                        createRequest(stack, false);
+                        itemStack.setCount(Math.min(itemStack.getMaxStackSize(), delta));
+                        final Stack stack = new Stack(itemStack);
```

Comment:
Pass in the delta here to do this properly. The Stack requestable can tract counts.

Reply:
I'm not doing this on purpose. I want to request 1 by 1